src/popup/components/InfiniteScroll.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <div
+    ref="scrollArea"
+    class="infinite-scroll"
+    @scroll="onScroll"
+  >
+    <slot />
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  defineComponent, onMounted, onUnmounted,
+} from '@vue/composition-api';
+
+export default defineComponent({
+  name: 'InfiniteScroll',
+  props: {
+    isMoreData: { type: Boolean, required: true },
+  },
+  setup(props, { emit }) {
+    const appInner = document.querySelector('.app-inner')!;
```

Comment:
I know that this idea of selecting the `app-inner` is used in few places, but I consider this as a really bad pattern.
What I suggest is to create a composable that would provide a bridge between the app viewport and other components.

Only an idea:
```
// App.vue
setup(props) {
  // Template Refs: https://github.com/vuejs/composition-api#limitations
  const innerElem = ref(null);

  const { initViewport } = useViewport();
  
  onMounted(() => {
    initViewport(innerElem);
  });
}
```

```
// Any other component, like the current one:
setup() {
  const { onViewportScroll } = useViewport();
  onViewportScroll(({ isDesktop ) => {
    // Some logic
  });
}
```